src/js/components/locations-configuration/SuccessMessage.jsx

```diff
@@ -36,7 +36,13 @@ const SuccessMessage = ({ setShowSuccessMessage, history, successMessageOpen })
             />
           </p>
           <div className="success-buttons-section d-flex flex-column">
-            <button type="button" className="btn btn-outline-primary" onClick={() => history.push(LOCATION_CONFIGURATION_URL.create())}>
+            <button
+              type="button"
+              className="btn btn-outline-primary"
+              onClick={() => {
+                window.location = LOCATION_CONFIGURATION_URL.create();
```

Comment:
have not noticed it before - why has it been changed from `history.push` to `window.location`? It's a React page, so it smells for me a bit :thinking: not knowing the reason for that change.

Reply:
The issue is that at this point in our application we are on URL `/openboxes/locationsConfiguration/create/:someId` and when trying to execute `history.push(LOCATION_CONFIGURATION_URL.create())` we are trying to redirect to `/openboxes/locationsConfiguration/create` but upon executing this history.push, nothing happens since from the point of view of react-router page doesn't change.

I assume, the reason for that is that we do not have a route defined for `/openboxes/locationsConfiguration` but only for `/openboxes/locationsConfiguration/:locationId` and we handle the logic ourselves for **with** and **without** id.

![image](https://github.com/openboxes/openboxes/assets/35233347/93248871-8af4-4276-9a9b-9980c9b4702a)

This issue will need to be investigated separately and is most likely a tech-debt.
My suggestion would be to explicitly define separate pages/steps in the browser router instead of relying on a single page with internal state tracking the page like id or step number.

For now all I can do is rely on window.location to completely refresh the page and redirect to `/openboxes/locationsConfiguration/create`